src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.java

```diff
@@ -331,7 +331,7 @@ else if (!hasEmptyLineAfter(ast)) {
      */
     private void processMultipleLinesInside(DetailAST ast) {
         final int astType = ast.getType();
-        if (astType != TokenTypes.CLASS_DEF && isClassMemberBlock(astType)) {
+        if (isClassMemberBlock(astType)) {
```

Comment:
Code removed should be valid. Removed code is `ast != Class` while statement inside method is `ast == ...`, so it is redundant to have them both.

`isClassMemberBlock` mutation is still failing and I want to look into this once more.

Reply:
`isClassMemberBlock` survives because when it returns `true` instead of `false`, `getEmptyLines` always returns an empty list and never produces a violation.
`ast` are PACKAGE_DEF, CLASS_DEF and VARIABLE_DEF.

For PACKAGE_DEF, `getEmptyLines` always returns an empty list no matter what.
For CLASS_DEF, `getEmptyLines` can return members and produce violations but if all tokens are set, it just duplicates violations found in METHOD_DEF.
So to kill the mutation, we must use only the CLASS_DEF token. I will add this test case. No reversion of code is needed.

Reply:
confirmed.